nixos/modules/system/activation/top-level.nix

```diff
@@ -299,6 +299,32 @@ in
       '';
     };
 
+    system.diagnostics.prefix = mkOption {
+      type = types.str;
+      default = "";
+      example = "in containers.foo: ";
+      description = ''
+        A prefix that is automatically added in the effect of the `warnings` and `assertions` options.
+
+        Knowning that a warning comes from a specific container can make all the difference, for instance.
+
+        How to provide good diagnostic context:
+        1.  If possible, either
+            - Use the `warnings` and `assertions` options if it's ok and easy to evaluate them out of band.
+            - Use `lib.warn` and refer to the option by `"''${options.myOption}"`, which includes a fully qualified path, like ${options.system.diagnostics.prefix}'s default value.
+        2.  Otherwise, use `lib.warn` or `throw` directly, and include the context in the message.
+            Example:
+            ```nix
+            lib.warn
+              "''${options.system.diagnostics.prefix} The options myOption and myOtherOption may conflict with each other. Check this  and perhaps do that."
+              foo
+            ```
+
+        Instead of setting ${options.system.diagnostics.prefix} directly, it is generally preferable to set the `prefix` parameter in the NixOS entrypoint, or (for this purpose) equivalently, to load NixOS into a submodule.
```

Comment:
This sentence isn't clear to me as it related to `prefix`. Where exactly does that get set? Maybe an example here would be clearer.

Reply:
It's an argument in nixos/lib/eval-config.nix`

https://github.com/NixOS/nixpkgs/blob/e79b71782a4c64e1d01d1befde483565a096f6e3/nixos/lib/eval-config.nix#L31